src/lifecycleEvents.ts

```diff
@@ -32,14 +36,20 @@ declare const global: {
  *
  * // Deep in the deploy code, fire the event for all libraries and plugins to hear.
  * Lifecycle.getInstance().emit('deploy-metadata', metadataToBeDeployed);
+ *
+ * // if you don't need to await anything
+ * use `void Lifecycle.getInstance().emit('deploy-metadata', metadataToBeDeployed)` ;
  * ```
  */
 export class Lifecycle {
+  public static readonly telemetryEventName = 'telemetry';
+  public static readonly warningEventName = 'warning';
   private debug = Debug(`sfdx:${this.constructor.name}`);
-  private readonly listeners: Dictionary<callback[]>;
 
-  private constructor() {
-    this.listeners = {};
+  private constructor(private readonly listeners: Dictionary<callback[]> = {}) {}
+
+  public static staticVersion(): string {
+    return pjson.version;
```

Comment:
Should this be private?

Reply:
I don't see the harm in public, and it makes the tests much simpler.

Reply:
I'm not concerned about that, but I did use it in the tests.